blog/_posts/2019-11-13-affiliations-feature.md

```diff
@@ -0,0 +1,67 @@
+
+Cleaning the data turned out to be more of an art than a science.  There exist curation tools for cleaning up messy data - the one we used almost exclusively was Open Refine (called Google Refine at the time).  This allowed bulk substitutions, expansions, translations and more.  But cleaning is not enough.  For example, changing all instances of “UC Irvine” to “University of California Irvine” may seem like a good idea, except that new instances of “UC Irvine” are likely to continue coming in.
+		 	 	 		
+We were working with a dataset of approximately 34 million unique strings (about 4.5m astronomy and 29+ million physics). In the end, we were able to identify approximately 2.25 million affiliation strings, resulting in matching 93% of astronomy affiliations and 70% of physics affiliations to an institutional identifier.
+
+
+Automated Pipelines: assigning identifiers to specific author-affiliation-bibcode entities
+
+With the initial set of affiliation strings identified we, along with ADS backend programmer Stephen McDonald, designed an automated pipeline to add canonical affiliation identifiers to our database of 15 million references.  It’s a simple process.  First we normalize all of the strings with all uppercase letters, remove spaces and a subset of punctuation marks not needed for disambiguation, which results in a reduced number of matching strings; often, strings will be identical without punctuation, such as strings with “... U.S.A.” and “... USA”.  We then assign the strings to a dictionary, where the strings themselves are the keys, and the values are the identifiers.  If a normalized incoming string matches a key in the dictionary, we assign it the same aff_id, and the record with augmented affiliation data is sent back to our database of metadata.  The augmented affiliations are then passed to Solr and then to the ADS website with the next reindexing process.
+
+We’ve entirely automated the process, so once the dictionary of affiliation strings and matched identifiers is created, it’s a hands-off component of our metadata processing pipeline.  We update dictionaries about once a month, and occasionally more often if we have a large batch of incoming unmatched affiliation strings.  This process is similar to the first curation step, with the only difference being that we prioritize strings according to how many times they appear.  For example, if we receive a number of papers from collaborations with hundreds or thousands of authors *and* their affiliations don’t exactly match our dictionary, we need to add them.  Early on in this process, we came across strings that had many thousands (and sometimes tens of thousands) of occurrences, but the latest batch of unclassified strings appear fewer than 800 times in our entire database and this maximum frequency drops each time we create a new dictionary.
```

Comment:
Pull the last bit out into its own sentence for emphasis: "This maximum frequency drops each time we create a new dictionary." This is impressive!